iree/turbine/kernel/wave/codegen.py

```diff
@@ -1205,13 +1283,77 @@ def handle_div(lhs: Value, rhs: Value) -> OpResult:
     return result
 
 
+@handle_binary_op(operator.gt)
```

Comment:
Note: originally, I intentionally didn't overload comparison operators. In my previous experience with python bindings and MLIR, this only brings pain and suffering.

Also, arith dialect operations are only expected to work on _signless_ integers, and will likely fail verificaiton or assert if you try feeding them signed/unsigned, so using the signedness bit to differentiate comparison types is most likely not going to work. It is just not exercised anywhere.

Reply:
@raikonenfnu +1 on not being too smart here and bias towards more explicit as we are mixing sympy, tracing and tkw functions.

Reply:
I think we should have support for both >,<, ... and tkw.gt,le etc. One of the big plusses was the readability and my two cents are that making it more readable is a good thing.

Reply:
Can you add an assertion to disallow unsigned types?

Reply:
Done and done!